### PR TITLE
Allow mmCIF parser to read auth_* records for residue numbers and chain IDs.

### LINKED ISSUE #1776: how to get label residue numbering instead of author numbering in MMCIFParser?
### Setup

```$ python
Python 2.7.15 (default, May 16 2018, 17:50:09) 
[GCC 8.1.1 20180502 (Red Hat 8.1.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys; print(sys.version)
2.7.15 (default, May 16 2018, 17:50:09) 
[GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
Linux-4.17.17-300.Lyude.bz1477182.V5.fc28.x86_64-x86_64-with-fedora-28-Twenty_Eight
>>> import Bio; print(Bio.__version__)
1.72
```

### Expected behaviour

I would like to have label sequence numbering instead of author sequence numbering for residues when MMCIFParser is used. I understand that can not be a default, but have not found a way how to achieve that.

### Actual behaviour

Only author sequence numbering for residues is returned.

Would you consider path like this?

Add something like following into `Bio.PDB.StructureBuilder`:

```
SEQ_ID_FIELD = '_atom_site.auth_seq_id'
ASYM_ID_FIELD = '_atom_site.auth_asym_id'
```

so you can easily change that by subclassing from that builder. Then in the `Bio.PDB.MMCIFParser` we would of course need:

```
-seq_id_list = mmcif_dict["_atom_site.auth_seq_id"]
+seq_id_list = mmcif_dict[self._structure_builder.SEQ_ID_FIELD]
```

Also we would need some default in the parser so any existing builders would still work.

Or did I got the whole concept wrong?

I need to use "label" numbering because I'm integrating with some other tool which ises that.

### PR DESCRIPTION
**Pull request by sjdv1982**
- X I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- X I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- X I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #1776 
Changes the parsers in MMCIFParser.py, adding two `__init__` parameters `auth_residues` and `auth_chains`, `True` by default. If disabled, "label_" records instead of "auth_" records are used for residue ID and chain ID, respectively.

### CODE REVIEW

File: Bio/PDB/MMCIFParser.py [hunk 1]
```diff
@@ -209,7 +228,11 @@ def _build_structure(self, structure_id):
             altloc = alt_list[i]
             if altloc in _unassigned:
                 altloc = " "
-            int_resseq = int(seq_id_list[i])
+            resseq = seq_id_list[i]
+            if resseq == ".":
+                # Non-existing residue ID
```
Comment (JoaoRodrigues) [inline Bio/PDB/MMCIFParser.py:233]: I would issue a warning for each residue that is skipped, so that the user knows that this is the case.

**Review by JoaoRodrigues — APPROVED** [review]
Minor point about adding a warning, but otherwise, 👍  for me. Thanks @sjdv1982 !

**Comment by sjdv1982** [comment]
You're welcome! Warning has been added.

**Comment by JoaoRodrigues** [comment]
Thanks Sjoerd! I'll take it from here, seems like there are some issues with Sphinx but that's unrelated to this PR.

**Comment by peterjc** [comment]
I think #3848 avoids the Sphinx issue, so a rebase/merge ought to fix that. Although a real fix would be welcome of course.

**Comment by sjdv1982** [comment]
I don't know much about git rebase, but I could merge master into this branch without conflicts. Will that be okay?

**Comment by peterjc** [comment]
Joao said leave it with him, two people editing the branch at the same time can be trouble.

However, what I would have tried is something like this (assuming you cloned the original repository as origin, and named you fork based on your username):

```
$ git fetch origin
$ git rebase origin/master
$ git push sjdv1982 issue1776 --force # rare occasion when should use force
```

Unless your branch edits files also edited on the master, this should just work without any merge conflicts - and it keeps the history simpler than a merge would do, but does re-write the history to make it look like the branch started from the current origin/master.

**Comment by sjdv1982** [comment]
@peterjc . Ah, thanks for the explanation, I still have a lot to learn about Git!

I didn't realize that pull requests could be edited as normal branches by maintainers. Before your comment, I thought that Joao would create a new branch owned by him, and then cherrypick my commits.

**Comment by peterjc** [comment]
That's a GitHub feature, by default by making a pull request you give write permissions *to that branch only* to the team of the upstream repository. i.e. People like me and Joao. As you guessed, this avoids sister branches and near duplicate pull requests. I find it a useful improvement - but it still helps to be clear about who might or might not be about to change it.

**Comment by sjdv1982** [comment]
Understood. I won't touch [sjdv1982:issue1776](https://github.com/sjdv1982/biopython/tree/issue1776) , feel free to rewrite its commit history.

**Comment by JoaoRodrigues** [comment]
Thanks @peterjc, I had a vague recollection of seeing this error before - now I remember that PR - that's why I offered to fix it for @sjdv1982. I'll rebase/merge today!

**Pull request merged**